ci/build_manylinux_wheel.sh

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env bash
+# Usage:
+#   ./build_manylinux_wheel.sh --os <matrix.os> --python <python-version> \
+#     --arch <ARCH> [--workspace <path>] [--x86-image <image>] \
+#     [--aarch64-image <image>] [--docker-image <image>] [--dry-run]
+#
+# Examples:
+#   ./build_manylinux_wheel.sh --os ubuntu-latest --python 3.10 --arch X64
+#   ./build_manylinux_wheel.sh --os ubuntu-24.04-arm --python 3.11 --arch ARM64 \
+#     --aarch64-image quay.io/pypa/manylinux_2014_aarch64:latest
+#
+# Notes:
+#   --arch accepts values: X86, X64, ARM, or ARM64 (case-insensitive).
+#   This script requires --arch to be provided explicitly.
+set -euo pipefail
+
+print_usage() {
+  cat <<EOF
+Usage: $0 --os <matrix.os> --python <python-version> --arch <ARCH> [options]
+
+Required:
+  --os                matrix.os value (e.g. ubuntu-latest or ubuntu-24.04-arm)
+  --python            Python version (e.g. 3.10)
+  --arch              Architecture (X86, X64, ARM, or ARM64)
+
+Optional:
+  --workspace         Path to workspace to mount into container (default: cwd)
+  --x86-image         manylinux x86_64 docker image (overrides default env)
+  --aarch64-image     manylinux aarch64 docker image (overrides default env)
+  --docker-image      Explicit docker image to use (skips auto selection)
+  --dry-run           Print the docker command without executing it
+  -h, --help          Show this help
+EOF
+}
+
+# Defaults - can be overridden by options
+WORKSPACE="${GITHUB_WORKSPACE:-$(pwd)}"
+MANYLINUX_X86_64_IMAGE="${MANYLINUX_X86_64_IMAGE:-quay.io/pypa/manylinux_2_28_x86_64:latest}"
```

Comment:
I take a look as pyarrow/numpy/opendal released files on pypi:

- https://pypi.org/project/numpy/1.26.4/#files
- https://pypi.org/project/numpy/2.2.6/#files
- https://pypi.org/project/numpy/#files
- https://pypi.org/project/pyarrow/#files
- https://pypi.org/project/opendal/#files

We may also needs to add `musllinux_1_2`.  pyarrow and numpy stoped  `manylinux2014`, seems it's ok to drop that. But according to https://github.com/pypa/manylinux, it seems we can get better compatibility if we also release manylinux2014 wheels.

One thing i found strange is that `auditwheel` give wrong name `pyfory-0.12.0a9-cp310-cp310-manylinux2014_x86_64.manylinux_2_17_x86_64.whl`, it should be `pyfory-0.12.0a9-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl`